### PR TITLE
fix DefaultValueResponce can't access DialogContext.State 

### PR DESCRIPTION
**Pull request by cosmicshuai**
closes: #4409 

I have an E2E test for this fix: Here is the dialog file

```JSON
{
  "$schema": "../../testbot.schema",
  "$kind": "Microsoft.AdaptiveDialog",
  "triggers": [
    {
      "$kind": "Microsoft.OnBeginDialog",
      "actions": [
        {
          "$kind": "Microsoft.SendActivity",
          "$designer": {
            "id": "859266",
            "name": "Send a response"
          },
          "activity": "Welcome User!"
        },
        {
          "$kind": "Microsoft.NumberInput",
          "defaultLocale": "en-us",
          "disabled": false,
          "maxTurnCount": 2,
          "alwaysPrompt": true,
          "allowInterruptions": false,
          "prompt": "Give me your favorite number (1-10)",
          "unrecognizedPrompt": "Sorry, ${turn.activity.text} did not include a valid number.",
          "invalidPrompt": "Sorry, ${this.value} does not work. Can you give me a different number that is between 1-10?",
          "defaultValueResponse": "Sorry, we have tried for ${class.MaxTurnCount} number of times and I'm still not getting it. For now, I'm setting ${class.property} to ${class.DefaultValue}",
          "property": "user.favoriteNumber",
          "outputFormat": "=string(this.value)",
          "validations": [
            "=int(this.value) >=1",
            "=int(this.value) <= 10"

          ],
          "defaultValue": 9
        },
        {
          "$kind": "Microsoft.SendActivity",
          "activity": "${user.favoriteNumber}"
        }
      ]
    }
  ]
}
```
The result is:
image

### CODE REVIEW

**Comment by boydc2014** [comment]
@cosmicshuai can you just check in the E2E test somewhere?

**Comment by cosmicshuai** [comment]
@boydc2014
I did the E2E test. 
I created a bot in Composer, and run it in the Testbot.json. 
I can upload the project somewhere, and append in the description.

**Comment by boydc2014** [comment]
> @boydc2014
> I did the E2E test.
> I created a bot in Composer, and run it in the Testbot.json.
> I can upload the project somewhere, and append in the description.

I mean in the test projects, we have UT which can test a full declarative dialog, right? can we check in a new test case to ensure this behavior is right?

File: libraries/Microsoft.Bot.Builder.Dialogs.Adaptive/Input/InputDialog.cs [hunk 1]
```diff
@@ -266,7 +266,7 @@ public override async Task<DialogTurnResult> ContinueDialogAsync(DialogContext d
                     var (value, error) = this.DefaultValue.TryGetValue(dc.State);
                     if (this.DefaultValueResponse != null)
                     {
-                        var response = await this.DefaultValueResponse.BindAsync(dc, cancellationToken).ConfigureAwait(false);
+                        var response = await this.DefaultValueResponse.BindAsync(dc, cancellationToken: cancellationToken).ConfigureAwait(false);
```
Comment (boydc2014) [inline libraries/Microsoft.Bot.Builder.Dialogs.Adaptive/Input/InputDialog.cs:269]: https://github.com/microsoft/botbuilder-dotnet/blob/4677d4ec6c5bd3659ec834d4c175f3036208e363/libraries/Microsoft.Bot.Builder.Dialogs/Prompts/ActivityPrompt.cs#L65

see here, can we defense explicitly on cancellationToken in BindAsync

Comment (cosmicshuai) [reply libraries/Microsoft.Bot.Builder.Dialogs.Adaptive/Input/InputDialog.cs:269]: added explicit type checking for passing cancellationToken as data in BindAsync

**Comment by EricDahlvang** [comment]
Please also add cancellationToken to the EndDialogAsync call in InputDialog's ContinueDialogAsync method:

`return await dc.EndDialogAsync().ConfigureAwait(false);`

image

Thanks!

**Comment by cosmicshuai** [comment]
@EricDahlvang 
cancellationToken in EndDialogAsync has been added

**Review by boydc2014 — APPROVED** [review]

**Pull request merged**